net/advanced-operations/securing-and-signing/sign-pdf-document-from-smart-card/_index.md

```diff
@@ -136,37 +139,70 @@ using (PdfFileSignature pdfSign = new PdfFileSignature(new Document(dataDir + "e
-    store.Open(System.Security.Cryptography.X509Certificates.OpenFlags.ReadOnly);
-    // Manually chose the certificate in the store
-    System.Security.Cryptography.X509Certificates.X509Certificate2Collection sel = System.Security.Cryptography.X509Certificates.X509Certificate2UI.SelectFromCollection(store.Certificates, null, null, System.Security.Cryptography.X509Certificates.X509SelectionFlag.SingleSelection);
-
-    Aspose.Pdf.Forms.ExternalSignature externalSignature = new Aspose.Pdf.Forms.ExternalSignature(sel[0]);
-    pdfSign.SignatureAppearance = dataDir + "demo.png";
-    pdfSign.Sign(1, "Reason", "Contact", "Location", true, new System.Drawing.Rectangle(100, 100, 200, 200), externalSignature);
-    pdfSign.Save(dataDir + "externalSignature2.pdf");
+    using (Facades.PdfFileSignature pdfSign = new Facades.PdfFileSignature())
+    {
+        pdfSign.BindPdf(document);
+
+        // Sign with certificate selection in the windows certificate store
+        System.Security.Cryptography.X509Certificates.X509Store store = new System.Security.Cryptography.X509Certificates.X509Store(System.Security.Cryptography.X509Certificates.StoreLocation.CurrentUser);
+        store.Open(System.Security.Cryptography.X509Certificates.OpenFlags.ReadOnly);
+        // Manually chose the certificate in the store
+        System.Security.Cryptography.X509Certificates.X509Certificate2Collection sel = System.Security.Cryptography.X509Certificates.X509Certificate2UI.SelectFromCollection(store.Certificates, null, null, System.Security.Cryptography.X509Certificates.X509SelectionFlag.SingleSelection);
+
+        Aspose.Pdf.Forms.ExternalSignature externalSignature = new Aspose.Pdf.Forms.ExternalSignature(sel[0]);
+        pdfSign.SignatureAppearance = dataDir + "demo.png";
+        pdfSign.Sign(1, "Reason", "Contact", "Location", true, new System.Drawing.Rectangle(100, 100, 200, 200), externalSignature);
+        pdfSign.Save(dataDir + "externalSignature2.pdf");
+    }
 }
 
-using (Facades.PdfFileSignature pdfSign = new Facades.PdfFileSignature(new Document(dataDir + "externalSignature2.pdf")))
+using (Document document = new Document(dataDir + "externalSignature1.pdf"))
 {
-    IList<string> sigNames = pdfSign.GetSignNames();
-    for (int index = 0; index <= sigNames.Count - 1; index++)
+    using (Facades.PdfFileSignature pdfSign = new Facades.PdfFileSignature(document))
     {
-        if (!pdfSign.VerifySigned(sigNames[index]) || !pdfSign.VerifySignature(sigNames[index]))
+        IList<string> sigNames = pdfSign.GetSignNames();
+        for (int index = 0; index <= sigNames.Count - 1; index++)
         {
-            throw new ApplicationException("Not verified");
+            if (!pdfSign.VerifySigned(sigNames[index]) || !pdfSign.VerifySignature(sigNames[index]))
```

Comment:
You should use  pdfSign.VerifySignature(sigNames[index]) only.  The VerifySigned is deprecated.
The VerifySigned is equivalent  of VerifySignature.